src/client.js

```diff
@@ -87,11 +90,18 @@ class Client extends Base {
         callback = extraHeaders;
         extraHeaders = options;
         options = temp;
+      } else if (typeof callback === 'object') {
+        extraHeaders = options;
+        options = callback;
+        callback = undefined;
+      }
+      if (!callback) {
```

Comment:
nitpick feel free to ignore: I think the code below can be written in a conciser way: 

```js
callback = callback || utils.createPromiseCallback();
self._invode(operation, args, location, callback, options, extraHeaders);
return callback.promise;
```

Reply:
+1